pkg/webhook/parser.go

```diff
@@ -279,37 +296,92 @@ func (Generator) Generate(ctx *genall.GenerationContext) error {
 		}
 	}
 
-	var objs []interface{}
-	if len(mutatingCfgs) > 0 {
-		objs = append(objs, &admissionreg.MutatingWebhookConfiguration{
-			TypeMeta: metav1.TypeMeta{
-				Kind:       "MutatingWebhookConfiguration",
-				APIVersion: admissionreg.SchemeGroupVersion.String(),
-			},
-			ObjectMeta: metav1.ObjectMeta{
-				Name: "mutating-webhook-configuration",
-			},
-			Webhooks: mutatingCfgs,
-		})
+	webhookVersions := g.WebhookVersions
+	if len(webhookVersions) == 0 {
+		webhookVersions = []string{defaultVersion}
+	}
+	mutatingWebhookConfiguration := admissionregv1.MutatingWebhookConfiguration{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "MutatingWebhookConfiguration",
+			APIVersion: admissionregv1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "mutating-webhook-configuration",
+		},
+		Webhooks: mutatingCfgs,
 	}
 
-	if len(validatingCfgs) > 0 {
-		objs = append(objs, &admissionreg.ValidatingWebhookConfiguration{
-			TypeMeta: metav1.TypeMeta{
-				Kind:       "ValidatingWebhookConfiguration",
-				APIVersion: admissionreg.SchemeGroupVersion.String(),
-			},
-			ObjectMeta: metav1.ObjectMeta{
-				Name: "validating-webhook-configuration",
-			},
-			Webhooks: validatingCfgs,
-		})
-
+	validatingWebhookConfiguration := admissionregv1.ValidatingWebhookConfiguration{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ValidatingWebhookConfiguration",
+			APIVersion: admissionregv1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "validating-webhook-configuration",
+		},
+		Webhooks: validatingCfgs,
 	}
 
-	if len(objs) > 0 {
-		return ctx.WriteYAML("manifests.yaml", objs...)
+	versionedWebhooks := make([][]interface{}, len(webhookVersions))
+	for i, ver := range webhookVersions {
+		if len(mutatingCfgs) > 0 {
+			objRaw := mutatingWebhookConfiguration.DeepCopy()
```

Comment:
These don’t need to be copied.

Reply:
This is needed I think, because we are modifying the webhookConfiguration in the for loop, for example, we remove the `AdmissionVersions` for `v1beta1`, and if `v1beta1` comes first before `v1`, then `AdmissionVersions` is removed in the first iteration, it will break the generation for `v1`

Reply:
Ah yes you're right.